### PR TITLE
new module netapp csv pools for aws

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
New Module: netapp CSV pools for aws

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
- aws_netapp_cvs_pool.py 

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: test/units/modules/cloud/amazon/test_aws_netapp_cvs_pool.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        }
+        get_post_api.return_value = my_pool, "Error"
+        get_aws_api.return_value = None, False
+        with pytest.raises(AnsibleFailJson) as exc:
+            my_obj.apply()
+        print('Info: test_create_aws_netapp_cvs_pool_fail: %s' % repr(exc.value))
+        assert exc.value.args[0]['msg'] is not None
+
+    @patch('ansible.modules.cloud.amazon.aws_netapp_cvs_pool.NetAppAWSCVS.get_aws_netapp_cvs_pool')
+    @patch('ansible.module_utils.netapp.AwsCvsRestAPI.post')
```
Comment (lonico) [inline test/units/modules/cloud/amazon/test_aws_netapp_cvs_pool.py]: post needs to be changed to delete here.

File: lib/ansible/modules/cloud/amazon/aws_netapp_cvs_pool.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    def __init__(self):
+        """
+        Parse arguments, setup state variables,
+        """
+        self.argument_spec = netapp_utils.aws_cvs_host_argument_spec()
+        self.argument_spec.update(dict(
+            state=dict(required=True, choices=['present', 'absent']),
+            region=dict(required=True, type='str'),
+            name=dict(required=True, type='str'),
+            new_name=dict(required=False, type='str'),
```
Comment (lonico) [inline lib/ansible/modules/cloud/amazon/aws_netapp_cvs_pool.py]: don't use new_name.   It should be from_name to maintain idempotency

**Review by lonico — CHANGES_REQUESTED** [review]

**Review by lonico — APPROVED** [review]
shipit

**Review by lonico — APPROVED** [review]
shipit

**Comment by anshulbehl** [comment]
All looks good, shipit

**Review by thedoubl3j — APPROVED** [review]
everything looks good, docs fragment in plac and tests added. shipit.

**Pull request merged**